src/data/types/future.ts

```diff
@@ -1,5 +1,15 @@
-export class Future {
-	constructor(readonly inner: string) {}
+import { Value } from "../value";
+
+export class Future extends Value {
+	constructor(readonly inner: string) {
+		super();
+	}
+
+	equals(other: unknown): boolean {
+		if (!(other instanceof Future)) return false;
+		// TODO ignore whitespace
```

Comment:
that is dangerous haha, strings need to be unaffected

Reply:
It was more so with the thought of ql-wasm comparison, however since I don't see that being viable I'll just remove the comment for now